Sources/HAP/Server/Device.SetupCode.swift

```diff
@@ -0,0 +1,67 @@
+import Foundation
+import Regex
+
+extension Device {
+    public enum SetupCode:Equatable {
+        case automatic
+        case `default`(String)
+        
+        // HAP Specification lists certain setup codes as invalid
+        public var isValid: Bool {
+            switch (self) {
+            case .default(let setupCode):
```

Comment:
Maybe we could change `automatic` into `random` and `default` into `predefined`? I think that would make the names more straight-forward.

Reply:
sure random is good, I don't like pre-defined because to me it implies that it would override what is in the configuration, which is not the case. Any other suggestion ?